clients/src/main/java/org/apache/kafka/common/utils/Utils.java

```diff
@@ -156,7 +156,7 @@ public static int utf8Length(CharSequence s) {
      * Read the given byte buffer into a byte array
      */
     public static byte[] toArray(ByteBuffer buffer) {
-        return toArray(buffer, 0, buffer.limit());
```

Comment:
The reason why `0` is being passed is that `position` is called in the overloaded `toArray` method. So, this is actually doing `2` although it's not too clear.